examples/mistral-4-node-benchmark.yaml

```diff
@@ -0,0 +1,60 @@
+training:
+  train_iters: 1000
+  num_workers: 8
+  logs:
+    interval: 10
+  checkpoint:
+    interval: 1000
+    keep: 1
+  export:
+    interval: 1000
+  validation:
+    iterations: null
+  test_iters: 0
+batch:
+  sequence_length: 8192
+  micro_batch_size: 1
+  batch_size: 32
+data:
+  format: random
+  split: [1, 0, 0]
+optimizer:
+  learning_rate:
+    base: 1.0e-05
+    decay_style: constant
+    warmup_iterations: 0
+  weight_decay: 0.1
+  beta_1: 0.9
+  beta_2: 0.95
+model:
+  base_model:
+    transformer:
+      normalization:
+        type: rms_norm
+        epsilon: 1.0e-05
+      num_layers: 32
+      hidden_size: 4096
+      ffn_hidden_size: 14336
+      num_attention_heads: 32
+      head_groups: 8
+      add_linear_biases: false
+      use_rotary_embeddings: true
+      gated: true
+      activation_type: silu
+      triton_rotary: true
+      kv_channels: 128
+      rotary_embedding_scale: -9.210340371976184
+      window_size: 4096
+      init_method_std: 0.009021
+      attention_dropout: 0.0
+      hidden_dropout: 0.0
+    vocab_size: 32000
+    tie_word_embeddings: false
+  multi_stage:
+    zero_stage: 3
```

Comment:
Stage 2 should be enough

Reply:
I have only benchmarked with stage 3 so far. I'd like to keep it that way, since this is also our internal benchmark.

Reply:
That must explain the low throughput, It's supposed to be at least around 9800 tokens/s, at least that's what I was getting in my benchmarks.

Reply:
Ah, interesting. Let me try to reproduce!

Reply:
Job b48db977-754f-4969-b4cc-398fd16ddfa7
```
2024-10-21 13:48:07,151 [Rank 00] Training @ iteration     20/   100 | consumed samples:          640 | consumed tokens:        5,242,880 | batch size:  32 | step time: 832.51 ms | throughput: 467.35 tflop/s (model) | 475.27 tflop/s (hardware) | 9840.09 tokens/s/gpu | Memory allocated 18,998.57 MiB | max allocated 52,928.71 MiB | reserved 53,586.00 MiB | max reserved 53,586.00 MiB | global max reserved 53,586.00 MiB | learning rate: 1.000e-05 | loss scale:     1 | grad norm: 45.2334 | skipped iterations:   0 | nan iterations:   0 | average step time 2142.16 ms | remaining 0:02:51  | completion 2024-10-21 13:50:59 (20.00 %) | language model loss: 6.49286 | run: 0
[...]
2024-10-21 13:49:05,814 [Rank 00] Training @ iteration     90/   100 | consumed samples:        2,880 | consumed tokens:       23,592,960 | batch size:  32 | step time: 839.65 ms | throughput: 463.38 tflop/s (model) | 471.23 tflop/s (hardware) | 9756.50 tokens/s/gpu | Memory allocated 18,998.57 MiB | max allocated 52,928.71 MiB | reserved 53,586.00 MiB | max reserved 53,586.00 MiB | global max reserved 53,586.00 MiB | learning rate: 1.000e-05 | loss scale:     1 | grad norm: 8.5320 | skipped iterations:   0 | nan iterations:   0 | average step time 1127.85 ms | remaining 0:00:11  | completion 2024-10-21 13:49:17 (90.00 %) | language model loss: 0.05204 | run: 0
```
It starts at ~9850 and then drops a bit because of GPUs cooking.

Reply:
switched to zero 2 in this config and up'ed benchmark figure in README